villainary-react/app/products/ProductContainer.tsx

```diff
@@ -0,0 +1,48 @@
+'use client';
+
+import { Grid } from '@mui/material';
+import { useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { getData } from '../api/api-handler';
+import Wallet from '../components/Wallet';
+import { products as stubProducts } from '../constants/products';
+import { Product } from '../interfaces/Product';
+import { updateCurrentFunds } from '../state/userState.slice';
+import ProductCard from './ProductCard';
+
+export default function ProductContainer() {
+    const [products, setProducts] = useState<Product[]>([]);
+    const dispatch = useDispatch();
+    const handleClick = (value: number) => {
+        return () => {
+            return dispatch(updateCurrentFunds(-value));
+        };
+    };
```

Comment:
After using Angular for so long, I really dislike the callback pattern that react relies upon for separating concerns between parent and child components.  Event emitters in Angular have spoiled me.